core/notification_manager.cpp

```diff
@@ -43,6 +47,7 @@ void NotificationManager::refreshNotificationFile()
   // server side. The id mechanism also ensures that individual announcements
   // will be displayed to users for the same time spans independently of the
   // time client received announcement.
+#ifndef IDF_TARGET
```

Comment:
Also - instead of adding a ton of exclusions I feel like this file deserves a treatment similar to `HooksManager` - adding some sort of `DummyNotificationManager` (and an interface) and maybe even not compiling the full one entirely (using ifdef like this but for most of the file instead of a bunch of small ones)